src/plugins/data/common/datasets/types.ts

```diff
@@ -247,6 +247,13 @@ export interface Dataset extends BaseDataset {
   timeFieldName?: string;
   /** Optional language to default to from the language selector */
   language?: string;
+  /** Optional reference to the source dataset. Example usage is for indexed views to store the
+   * reference to the table dataset 
+   */
+  ref?: {
```

Comment:
Shouldn't we name this something less generic?

@kavilla Is this the right place for this? Isn't the source an attribute of the index that was created? Would expect that to be part of DataSource definition I guess, although if this is a two-way door seems fine for now (I think it would be as not having source in the right place wouldn't impact functionality for saved query/recent dataset?).

Reply:
that is a good point. for example we do store the session id wtih the meta info about the data source since it's the data source that decides that where as the data source would also include the index that is pointing to. originally i think we though it should go in the data source meta but didn't have too much of hard buy in on the data source. but reflecting it makes more sense there.

agreeed as well this is a two way door. as i believe we can still be able to query the info i think the ref now is used for ui purposes (for now) but def will need to update it.

Reply:
Can we create an issue to track the rename and move?